File: csharp/1040.cs
using System; 

class URI {

    static void Main(string[] args) { 

            string[] entrada = Console.ReadLine().Split(' ');
            float[] numeros = Array.ConvertAll(entrada, float.Parse);
            
            float
            n1 = numeros[0],
            n2 = numeros[1],
            n3 = numeros[2],
            n4 = numeros[3],
            resultado = (float)((n1*2+n2*3+n3*4+n4)/10.0f),
            notaArred = (float)Math.Round(resultado*10)/10.0f,
            notaExame = 0,
            mediaFinal = 0;
            
            Console.WriteLine($"Media: {notaArred:F1}");
            if (notaArred >= 7)
            {
                Console.WriteLine("Aluno aprovado.");
            }
            else if (notaArred < 5)
            {
                Console.WriteLine("Aluno reprovado.");
            }
            else
            {
                Console.WriteLine("Aluno em exame.");
                notaExame = float.Parse(Console.ReadLine());
                Console.WriteLine($"Nota do exame: {notaExame:F1}");
                mediaFinal = ((notaExame+notaArred)/2.0f);
                if(mediaFinal >= 5)
                {
                    Console.WriteLine("Aluno aprovado.");
                    Console.WriteLine($"Media final: {mediaFinal:F1}");
                }
                else
                {
                    Console.WriteLine("Aluno reprovado.");
                    Console.WriteLine($"Media final: {mediaFinal:F1}");
                }
            }

    }

}

File: java 19/1040.java
import java.io.IOException;
import java.util.Scanner;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);        
        double n1 = sc.nextDouble();
        double n2 = sc.nextDouble();
        double n3 = sc.nextDouble();
        double n4 = sc.nextDouble();
        sc.nextLine();
        int[] pesos = {2,3,4,1};
        int somaPesos = Arrays.stream(pesos).sum();
        double media = (n1*pesos[0]+n2*pesos[1]+n3*pesos[2]+n4*pesos[3])/somaPesos;
        media = Math.floor(media*10)/10;
        System.out.printf("Media: %.1f\n",media);
        if (media < 5.0) {
            System.out.println("Aluno reprovado.");
        } else if (media >= 7) {
            System.out.println("Aluno aprovado.");
        } else {
            System.out.println("Aluno em exame.");
            double nExame = sc.nextDouble();
            sc.nextLine();
            System.out.printf("Nota do exame: %.1f\n",nExame);
            double mediaFinal = (media + nExame)/2.0;
            String resultadoFinal = (mediaFinal >= 5.0) ? "Aluno aprovado." : "Aluno reprovado.";
            System.out.println(resultadoFinal);
            System.out.printf("Media final: %.1f\n", mediaFinal);
        }
        sc.close();
    }
}
